src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1057,917 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+    if kube_config:
+        cmd_helm_values.extend(["--kubeconfig", kube_config])
+    if kube_context:
+        cmd_helm_values.extend(["--kube-context", kube_context])
+
+    response_helm_values_get = Popen(cmd_helm_values, stdout=PIPE, stderr=PIPE)
+    output_helm_values, error_helm_get_values = response_helm_values_get.communicate()
+    if response_helm_values_get.returncode != 0:
+        if 'forbidden' in error_helm_get_values.decode("ascii"):
+            telemetry.set_user_fault()
+        telemetry.set_exception(exception=error_helm_get_values.decode("ascii"), fault_type=consts.Get_Helm_Values_Failed,
+                                summary='Error while doing helm get values azure-arc')
+        raise CLIError("Error while getting the helm values in the azure-arc namespace: " + error_helm_get_values.decode("ascii"))
+
+    output_helm_values = output_helm_values.decode("ascii")
+
+    try:
+        existing_values = yaml.safe_load(output_helm_values)
+        return existing_values
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Helm_Existing_User_Supplied_Value_Get_Fault,
+                                summary='Problem loading the helm existing values')
+        raise CLIError("Problem loading the helm existing values: " + str(e))
+
+
+def enable_features(cmd, client, resource_group_name, cluster_name, features, kube_config=None, kube_context=None,
+                    azrbac_client_id=None, azrbac_client_secret=None):
+    logger.warning("Ensure that you have the latest helm version installed before proceeding.")
+    logger.warning("This operation might take a while...\n")
+
+    features = [x.lower() for x in features]
+    enable_cluster_connect, enable_azure_rbac, enable_cl = utils.check_features_to_update(features)
+
+    if enable_azure_rbac:
+        if (azrbac_client_id is None) or (azrbac_client_secret is None):
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception='Client ID or secret not provided for Azure RBAC', fault_type=consts.Client_Details_Not_Provided_For_Azure_RBAC_Fault,
+                                    summary='Both client id and client secret is required to enable Azure RBAC feature')
+            raise CLIError("Please provide both client id and client secret to enable Azure RBAC feature")
+
+    if enable_cl:
+        enable_cl, custom_locations_oid = check_cl_registration_and_get_oid(cmd)
+        if not enable_cluster_connect and enable_cl:
+            enable_cluster_connect = True
+            features.append("cluster-connect")
+            logger.warning("Enabling 'custom-locations' feature will enable 'cluster-connect' feature too.")
+        if not enable_cl:
+            features.remove("custom-locations")
+            if len(features) == 0:
+                raise CLIError("Based on the inputs given and the warnings displayed above, there are no features to be enabled.")
```

Comment:
Let's set this as user fault before raising error

Reply:
done